chain/chain/src/store.rs

```diff
@@ -1983,39 +1983,50 @@ impl<'a> ChainStoreUpdate<'a> {
     pub fn clear_chunk_data_and_headers(
         &mut self,
         min_chunk_height: BlockHeight,
+        gc_limit: BlockHeightDelta,
+        archival: bool,
     ) -> Result<(), Error> {
         let chunk_tail = self.chunk_tail()?;
+        let min_chunk_height = std::cmp::min(min_chunk_height, chunk_tail.saturating_add(gc_limit));
         for height in chunk_tail..min_chunk_height {
             let chunk_hashes = self.chain_store.get_all_chunk_hashes_by_height(height)?;
             for chunk_hash in chunk_hashes {
                 // 1. Delete chunk-related data
-                let chunk = self.get_chunk(&chunk_hash)?.clone();
-                debug_assert_eq!(chunk.cloned_header().height_created(), height);
-                for transaction in chunk.transactions() {
-                    self.gc_col(ColTransactions, &transaction.get_hash().into());
-                }
-                for receipt in chunk.receipts() {
-                    self.gc_col(ColReceipts, &receipt.get_hash().into());
+                if !archival {
+                    let chunk = self.get_chunk(&chunk_hash)?.clone();
+                    debug_assert_eq!(chunk.cloned_header().height_created(), height);
```

Comment:
drive by, but

```suggestion
                    debug_assert_eq!(chunk.height_created(), height);
```

also, it feels like, rather than `debug_assert`, we should generally be using 

```
if !cond {
  if cfg(debug_assertions) { panic!() } else { tracing::error!() }
}
```

pattern, but this goes well beyond the scope of the PR